src/completions.jl

```diff
@@ -187,25 +165,100 @@ end
 completionicon(::REPLCompletions.DictCompletion) = "icon-key"
 completionicon(::REPLCompletions.PathCompletion) = "icon-file"
 
-function localcompletions(text, line, col)
-  ls = locals(text, line, col)
-  reverse!(ls)
-  map(localcompletion, ls)
+completiondetailtype(c) = ""
+completiondetailtype(::REPLCompletions.ModuleCompletion) = "module"
+completiondetailtype(::REPLCompletions.KeywordCompletion) = "keyword"
+completiondetailtype(c::REPLCompletions.MethodCompletion) = repr(hash(c.method)) # hash string to identify this method
+
+localcompletions(text, row, col, prefix) = localcompletion.(locals(text, row, col), prefix)
+localcompletion(l, prefix) = CompletionSuggetion(
+  :replacementPrefix  => prefix,
+  # suggestion body
+  :text               => l[:name],
+  :type               => l[:type] == "variable" ? "attribute" : l[:type],
+  :icon               => l[:icon] == "v" ? "icon-chevron-right" : l[:icon],
+  :rightLabel         => l[:root],
+  # for `getSuggestionDetailsOnSelect` API
+  :detailtype         => "", # shouldn't complete
+)
+
+### completion details on selection ###
+
+handle("completiondetail") do _comp
+  comp = Dict(Symbol(k) => v for (k, v) in _comp)
+  completiondetail!(comp)
+  return comp
+end
+
+function completiondetail!(comp)
+  isempty(comp[:detailtype]) && return comp
+
+  if comp[:detailtype] == "module"
+    completiondetail_module!(comp)
+  elseif comp[:detailtype] == "keyword"
+    completiondetail_keyword!(comp)
+  else # detail for method completion
+    completiondetail_method!(comp)
+  end
+
+  comp[:detailtype] = ""
+end
+
+function completiondetail_module!(comp)
+  mod = getmodule(comp[:rightLabel])
+  word = comp[:text]
+  cangetdocs(mod, word) || return
+  comp[:description] = completiondescription(getdocs(mod, word))
 end
 
-function localcompletion(l)
-  return Dict(
-    :type        => l[:type] == "variable" ? "attribute" : l[:type],
-    :icon        => l[:icon] == "v" ? "icon-chevron-right" : l[:icon],
-    :rightLabel  => l[:root],
-    :leftLabel   => "",
-    :text        => l[:name],
-    :description => ""
-  )
+function completiondetail_keyword!(comp)
+  comp[:description] = completiondescription(getdocs(Main, comp[:text]))
 end
 
-handle("cacheCompletions") do mod
-  # m = getthing(mod)
-  # m = isa(m, Module) ? m : Main
-  # CodeTools.completions(m)
+using JuliaInterpreter: sparam_syms
+using Base.Docs
+
+function completiondetail_method!(comp)
+  mod = getmodule(comp[:rightLabel])
+  # more robust way to do this ?
+  f = getfield′(mod, Symbol(comp[:replacementPrefix]))
+  isundefined(f) && return
```

Comment:
This will be more complex but maybe worth to create an additional key in baseline completions to keep a method's name from `Method.name`, since `prefix` is made in `REPL` module and may change in the future.

Reply:
nvm, addressed